design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
+- Scenario 1: <https://www.loom.com/share/def72e9ff7754b9e833c18dbdf9958c7>
+- Scenario 2: <https://www.loom.com/share/6aeeb12f776f4307beca97057ff60c48>
+- Scenario 3: <https://www.loom.com/share/d2cf81ac75ce47e8941669720ecc28e9>
+- Scenario 4: <https://www.loom.com/share/be12a128b41d443281babab0ccbcd112>
+- Scenario 5: <https://www.loom.com/share/9b5ab511981e4850acd7453fc907c81b>
 
-### Foreign Key
-The foreign key is a relationship between two tables that identifies which column in the other table references the first one.
+## Scenarios
 
-### Lookup Column
-They are used to show information from linked records in another table. The lookup column setting determines which column values are visible when displaying the record.
+### Scenario 1: User Adds a Foreign Key Constraint
 
-## Scenarios
-### A user wants to create a one-to-many table relationship between two existing tables
-A user has a table named 'artist' that contains records for artists in a music album collection. The user wants to link each artist to records in another table named 'track.' If successful, the user should link all tracks associated with an artist to records in the 'artist' table.
+A user wants to link records from another table into their current table. Both tables belong to the same schema. For this purpose, foreign key constraints can be applied to change the relationship between one table and another based on the values of columns of the linked tables.
 
+### Scenario 1a: Automatically adding foreign key constraints from the 'Link Table' dialog
 
-### Steps
+#### Steps for 1a
 
-#### 1. The user opens the 'track' table and adds a new column to link records from the 'artist' table
-- From the sidebar navigation list of tables, click on a table named 'track.'
-- Click on 'Add new Column,' and a new column is added with text type by default
-#### The user opens the new column menu and selects the 'Link to Another Table' option
-- Click on the column header to show the menu and select the 'Link to Another Table' option
+- The user starts the 'Link Table' process by clicking on the 'Link Table' button in the table toolbar area.
+- The user reads the instructions presented within the [link table](#link-table) component and understands that the tables will be linked by setting up a foreign key constraint. They also understand that manual configuration is available.
+- The user selects the table they wish to link to from the [table selector](#table-selector).
+- A list of questions is displayed. The user can answer 'yes' or 'no' depending on the relationship they want to create.
+- The user answers all the questions listed. The answers will determine the location of the foreign key or whether a new table needs to be created.
+  - Answering 'yes' to both questions will set up a mapping table with foreign key columns for both tables, creating a many-to-many relationship.
```

Comment:
We also need to handle the case where the answer is no for both questions. In this case, we can create the FK on the table the user is currently viewing and also set a unique constraint on that column so that each record can only be linked to another, unique, record.

Reply:
@kgodey I have updated both the prototype and spec to include the one-to-one scenario